pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -592,20 +595,31 @@ func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) erro
 
 func (sip *streamIngestionProcessor) flush() (*jobspb.ResolvedSpans, error) {
 	flushedCheckpoints := jobspb.ResolvedSpans{ResolvedSpans: make([]jobspb.ResolvedSpan, 0)}
+	if len(sip.curBatch) == 0 {
+		return &flushedCheckpoints, nil
+	}
+
 	// Ensure that the current batch is sorted.
 	sort.Sort(sip.curBatch)
 
 	totalSize := 0
+	minBatchMVCCTimestamp := hlc.MaxTimestamp
 	for _, kv := range sip.curBatch {
-		if err := sip.batcher.AddMVCCKey(sip.Ctx, kv.Key, kv.Value); err != nil { // problem is here
+		if err := sip.batcher.AddMVCCKey(sip.Ctx, kv.Key, kv.Value); err != nil {
 			return nil, errors.Wrapf(err, "adding key %+v", kv)
 		}
+		if kv.Key.Timestamp.Less(minBatchMVCCTimestamp) {
+			minBatchMVCCTimestamp = kv.Key.Timestamp
+		}
 		totalSize += len(kv.Key.Key) + len(kv.Value)
 	}
 
+	preFlushTime := timeutil.Now()
```

Comment:
looks like this won't record if flush returned an error?
do you think something like this would work:
```
defer func() {
  start := timeutil.Now()
  return func() {
   sip.metrics.FlushHistNanos.RecordValue(
     timeutil.Since(start).Nanoseconds())
  }
}()()